internal/core/attribute.cpp

```diff
@@ -30,33 +30,42 @@ Horizon::Window Horizon::at(std::size_t i) const noexcept (false) {
     };
 }
 
+void AttributeMap::AttributeComputeParam::update(
+    InputIt begin,
+    InputIt end,
```

Comment:
But that logic every function should be `T some_func(some_func_params const&)`. It's a poor design because it becomes less clear what arguments you expect (you have an indirection in order to figure it out). I see how it's _nessesary_ / nice in order to unify attribute::calculate signatures, but no need to over do it.